stacks_queues/queue.py

```diff
@@ -1,61 +1,93 @@
-
 INITIAL_QUEUE_SIZE = 20
 
+
 class QueueFullException(Exception):
     pass
 
+
 class QueueEmptyException(Exception):
     pass
 
-class Queue:
 
+class Queue:
     def __init__(self):
         self.store = [None] * INITIAL_QUEUE_SIZE
         self.buffer_size = INITIAL_QUEUE_SIZE
         self.front = -1
         self.rear = -1
         self.size = 0
-      
 
     def enqueue(self, element):
-        """ Adds an element to the Queue
-            Raises a QueueFullException if all elements
-            In the store are occupied
-            returns None
+        """Adds an element to the Queue
+        Raises a QueueFullException if all elements
+        In the store are occupied
+        returns None
         """
-        pass
+        if self.size == INITIAL_QUEUE_SIZE:
+            raise QueueFullException  # Overflow condition - if queue is full
+
+        self.size += 1
+        # last
+        if self.rear == (INITIAL_QUEUE_SIZE - 1):
+            self.rear = 0
+        else:
+            self.rear += 1
+
+        # front
+        if self.front == -1:
+            self.front = self.rear
+        self.store[self.rear] = element
 
     def dequeue(self):
-        """ Removes and returns an element from the Queue
-            Raises a QueueEmptyException if 
-            The Queue is empty.
+        """Removes and returns an element from the Queue
+        Raises a QueueEmptyException if
+        The Queue is empty.
         """
-        pass
+        if self.empty():
+            raise QueueEmptyException  # queue is empty
+
+        self.size -= 1
+        front = self.store[self.front]
+        self.store[self.front] = None
+
+        if self.front == (INITIAL_QUEUE_SIZE - 1):
+            self.front = 0
+        self.front += 1
+
+        return front
 
     def front(self):
-        """ Returns an element from the front
-            of the Queue and None if the Queue
-            is empty.  Does not remove anything.
+        """Returns an element from the front
+        of the Queue and None if the Queue
+        is empty.  Does not remove anything.
         """
-        pass
-        
+        self.store[self.front]
```

Comment:
The function specification asks you to return the element 

```suggestion
        return self.store[self.front]
```